### PR TITLE
Add examples on create index and training

### PR DESCRIPTION
**Pull request by prrao87**
Updates the Lance docs on HF with the following:
- [x] Show how to create an index if it doesn't yet exist in Lance
- [x] Point to `torchcodec` and training examples to show that Lance can help power HF users' training pipelines too.
- [x] Point to the new [Openvid-1M](https://huggingface.co/datasets/lance-format/Openvid-1M) dataset where video blob column is the first column in the list

### CODE REVIEW

**Comment by prrao87** [comment]
cc @lhoestq I realized I forgot to include the torchcodec tip you had in the docs. Please merge when you can, thanks!

File: docs/hub/datasets-lance.md [hunk 1]
```diff
@@ -219,6 +237,22 @@ with open("video_0.mp4", "wb") as f:
     f.write(blob_file.read())
 ```
 
+## Prepare data for training
+
+Training is another area where Lance's fast random access and scan performance can be useful. You can use Lance datasets as the storage mechanism for your training data, shuffling it and loadint into batches as part of your training pipelines.
```
Comment (julien-c) [inline docs/hub/datasets-lance.md]:
```suggestion
Training is another area where Lance's fast random access and scan performance can be useful. You can use Lance datasets as the storage mechanism for your training data, shuffling it and loading into batches as part of your training pipelines.
```

**Review by davanstrien — APPROVED** [review]
